src/url-mapper.js

```diff
@@ -4,18 +4,73 @@ export default class UrlMapper {
     this._db = db;
     this._update = update;
     this._map = {};
-    this._mapByNewUrl = {};
-    this._count = 0;
+    this._wildcards = {};
 
-    this.getList((err, mappedUrls) => {
+    this._db.find({}, (err, mappedUrls) => {
       mappedUrls.forEach((mappedUrl) => {
         this._addMemoryCopy(mappedUrl);
       });
+      update(); // Inform UI that mappings have been loaded
     });
   }
 
   get(url) {
-    return this._map[url];
+    const plainUrl = this._map[url];
+    if (plainUrl) {
+      return plainUrl;
+    }
+
+    const wildcardList = [];
+    for (const key in this._wildcards) {
+      if (this._wildcards.hasOwnProperty(key)) {
+        wildcardList.push(this._wildcards[key]);
```

Comment:
Might the following also be an option?

const wildcardList = Object.keys(this._wildcards).map((key) => this._wildcards[key]);